src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3395,6 +3395,8 @@ private int MinBatchSize
             var txAllocations = indexingContext.Transaction.InnerTransaction.LowLevelTransaction.NumberOfModifiedPages
                                 * Voron.Global.Constants.Storage.PageSize;
 
+            txAllocations += indexingContext.Transaction.InnerTransaction.LowLevelTransaction.TotalEncryptionBufferSize;
```

Comment:
@ayende  Regarding txAllocations limit in index: By default the value for Configuration.TransactionSizeLimit is null, therefore, we will not engage the logic that uses this value. In that case, should we have a default value for Index.TransactionSizeLimit in the case that there is no such value in the config? If we do, should it be 64MB, as we have in Bulk Insert?

Reply:
64MB is _very_ small. In the bulk insert case, we use that to limit the amount of data we read from the network.
The problem here is that we need to balance the speed of indexing with the cost of the transaction.

Reply:
I didn't notice that this is about encrypted databases.
For those scenarios, we don't want to keep too much in plain text in memory. We also have to worry about how much memory we can lock. So let's got with 64MB by default for encrypted dbs here.